pkg/sdk/tasks_test.go

```diff
@@ -0,0 +1,108 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testTasks struct {
+	tasks
+
+	stubbedTasks map[string]*Task
+}
+
+func (v *testTasks) ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Task, error) {
+	t, ok := v.stubbedTasks[id.Name()]
+	if !ok {
+		return nil, errors.New("no task configured, check test config")
+	}
+	return t, nil
+}
+
+func TestTasks_GetRootTasks(t *testing.T) {
+	db := "database"
+	sc := "schema"
+
+	setUpTasks := func(p map[string][]string) map[string]*Task {
+		r := make(map[string]*Task)
+		for k, v := range p {
+			if k == "initial" || k == "expected" {
+				continue
+			}
+			t := Task{DatabaseName: db, SchemaName: sc, Name: k}
+			predecessors := make([]SchemaObjectIdentifier, len(v))
+			for i, predecessor := range v {
+				predecessors[i] = NewSchemaObjectIdentifier(db, sc, predecessor)
+			}
+			t.Predecessors = predecessors
+			r[k] = &t
+		}
+		return r
+	}
+
+	tests := []map[string][]string{
```

Comment:
I have started with struct, but it was a bit more verbose. I prefer it now because these are all just easy-to-read one-liners. But I can show you how it looked before and you will decide if we keep it or leave it.

Reply:
Discussed. We leave the map with a introduction comment.